girder/utility/plugin_utilities.py

```diff
@@ -87,6 +87,30 @@ def loadPlugins(plugins, root, appconf, apiRoot=None, curConfig=None):
 
     return root, appconf, apiRoot
 
+def getPluginParentDir(name, curConfig=None):
+    """
+    Finds the directory a plugin lives in and returns it. This throws
+    an exception if it can't find a directory named name in any of the
+    set plugin_directory paths.
+
+    Also throws an exception if it finds multiple parents, i.e. the plugin
+    named name lives in more than one place.
+
+    :params name: The name of the plugin (i.e. its directory name)
+    :type name: str
+    """
+    parentDirs = set()
+
+    for potentialParentDir in getPluginDir(curConfig):
+        if os.path.exists(os.path.join(potentialParentDir, name)):
+            parentDirs.add(potentialParentDir)
+
+    if len(parentDirs) > 1:
+        raise Exception('Plugin directory %s exists in multiple paths' % name)
```

Comment:
We may want to allow the plugin to exist in multiple places and return the first (or last maybe) directory found.  I.e. something like the precedence of config file settings.

I could also be convinced that is a bad idea though...